### ISSUE TITLE
stolonctl: implement dead keeper removal

### ISSUE THREAD

**Issue by sgotti** [sorintlab/stolon#87]
Dead keeper are polluting the cluster data and the logs. I can see two different mode to remove them:
- [x] Add a 'DeadKeeperRemovalInterval' to the cluster spec, the sentinels will remove dead keepers after they are failed for more than the specified interval. A sane default could be something like 48 hours.
- [x] Add a stolonctl keeper remove command to manually remove a keeper. To be safer, stolonctl should check that the keeper is failed and not the current master before doing the removal (and perhaps add a `--force` option).

**Comment by sgotti** [comment]
Created #280 to implement the first point.

/cc @cuu508 if you want to try it out.

**Comment by sergik** [comment]
We are testing different production ready cases with stolon and faced following issue:
We remove keeper from the cluster. Sentinel starts following: 
"no keeper info available", zap.String("db", db.UID), zap.String("keeper", db.Spec.KeeperUID)
I found that there is mechanism for automatic keepers removal. But I do not see how it should work. You have the following:

```
for _, k := range newcd.Keepers {

			// get db associated to the keeper

			db := s.getDBForKeeper(cd, k.UID)

			if db != nil {

				// skip keepers with an assigned db

				continue

			}

			if (time.Now().After(k.Status.LastHealthyTime.Add(cd.Cluster.DefSpec().DeadKeeperRemovalInterval.Duration)){

				log.Info("removing old dead keeper", zap.String("keeper", k.UID))

				keepersToRemove = append(keepersToRemove, k)

				//remove db associated to keeper

				if db != nil {

					dbsToRemove = append(dbsToRemove, db)

				}

			}

		}
```

But looks like db is still in cluster config so we never try to remove dead keeper. if db != nil - always true. I added some tracing to this loop:

```
for _, k := range newcd.Keepers {
			// get db associated to the keeper
			db := s.getDBForKeeper(cd, k.UID)
			log.Info("db:")
			log.Info(fmt.Sprintln(db))
			log.Info("keeper:")
			log.Info(fmt.Sprintln(k))
			if db != nil {
				// skip keepers with an assigned db
				continue
			}
			log.Info("Time now:")
			log.Info(fmt.Sprintln(time.Now()))
			log.Info("Last healthy time:")
			log.Info(fmt.Sprintln(k.Status.LastHealthyTime))
			log.Info("Dead keeper remove interval:")
			log.Info(fmt.Sprintln(cd.Cluster.DefSpec().DeadKeeperRemovalInterval.Duration))
			log.Info("Evaluated:")
			log.Info(fmt.Sprintln(time.Now().After(k.Status.LastHealthyTime.Add(cd.Cluster.DefSpec().DeadKeeperRemovalInterval.Duration))))
			if time.Now().After(k.Status.LastHealthyTime.Add(cd.Cluster.DefSpec().DeadKeeperRemovalInterval.Duration)) {
				log.Info("removing old dead keeper", zap.String("keeper", k.UID))
				keepersToRemove = append(keepersToRemove, k)
				//remove db associated to keeper
				if db != nil {
					dbsToRemove = append(dbsToRemove, db)
				}
			}
		}
}
```
And here is the result:
![image](https://user-images.githubusercontent.com/358684/27539675-0e0209b8-5a86-11e7-93cb-be6ab765a0fb.png)
Could you please explain how it should work, is not but there?

**Comment by sgotti** [comment]
@sergik I'm on vacation so I can look at this carefully until next week but currently dead keepers are removed only if no db is assigned and a db won't be removed until a better keeper is found. So, depending on the max standby number you've defined in the cluster spec, you should add new keeper to have dead keepers removed (see the related integration tests). It could probably be improved in future and a command to stolonctl (point 2 of this issue) can be added to remove a keeper.

**Comment by sergik** [comment]
added a pull request for remove cmd https://github.com/sorintlab/stolon/pull/301

**Comment by sgotti** [comment]
fixed in #383
